src/hyp3_isce2/water_mask.py

```diff
@@ -59,36 +75,50 @@ def create_water_mask(input_image: str, output_image: str, gdal_format='GTiff'):
         output_image: Path for the output image
         gdal_format: GDAL format name to create output image as
     """
-    src_ds = gdal.Open(input_image)
-
-    driver_options = []
-    if gdal_format == 'GTiff':
-        driver_options = ['COMPRESS=LZW', 'TILED=YES', 'NUM_THREADS=ALL_CPUS']
-
-    dst_ds = gdal.GetDriverByName(gdal_format).Create(
-        output_image,
-        src_ds.RasterXSize,
-        src_ds.RasterYSize,
-        1,
-        gdal.GDT_Byte,
-        driver_options,
-    )
-    dst_ds.SetGeoTransform(src_ds.GetGeoTransform())
-    dst_ds.SetProjection(src_ds.GetProjection())
-    dst_ds.SetMetadataItem('AREA_OR_POINT', src_ds.GetMetadataItem('AREA_OR_POINT'))
 
-    envelope_wgs84_gdf = get_envelope_wgs84(input_image)
-
-    mask_location = '/vsicurl/https://asf-dem-west.s3.amazonaws.com/WATER_MASK/GSHHG/hyp3_water_mask_20220912.shp'
-
-    mask = gpd.read_file(mask_location, mask=envelope_wgs84_gdf)
-
-    mask = mask.clip(envelope_wgs84_gdf)
-
-    with TemporaryDirectory() as temp_dir:
-        temp_file = str(Path(temp_dir) / 'mask.shp')
-        mask.to_file(temp_file, driver='ESRI Shapefile')
-        with GDALConfigManager(OGR_ENABLE_PARTIAL_REPROJECTION='YES'):
-            gdal.Rasterize(dst_ds, temp_file, allTouched=True, burnValues=[1])
+    tiles = get_tiles(input_image, tmp_path=tmp_path)
+
+    if len(tiles) < 1:
+        raise ValueError(f'No water mask tiles found for {tiles}.')
+
+    tmp_px_size_path = tmp_path + 'tmp_px_size.tif'
+    merged_tif_path = tmp_path + 'merged.tif'
+    merged_vrt_path = tmp_path + 'merged.vrt'
+    merged_warped_path = tmp_path + 'merged_warped.tif'
+    shape_path = tmp_path + 'tmp.shp'
+
+    pixel_size = gdal.Warp(tmp_px_size_path, input_image, dstSRS='EPSG:4326').GetGeoTransform()[1]
+
+    # This is WAY faster than using gdal_merge, because of course it is.
+    if len(tiles) > 1:
+        build_vrt_command = ' '.join(['gdalbuildvrt', merged_vrt_path] + tiles)
+        system(build_vrt_command)
+        translate_command = ' '.join(['gdal_translate', merged_vrt_path, merged_tif_path])
+        system(translate_command)
+
+    shapefile_command = ' '.join(['gdaltindex', shape_path, input_image])
+    system(shapefile_command)
+
+    warp_filename = merged_tif_path if len(tiles) > 1 else tiles[0]
+
+    gdal.Warp(
+        merged_warped_path,
+        warp_filename,
+        cutlineDSName=shape_path,
+        cropToCutline=True,
+        xRes=pixel_size,
+        yRes=pixel_size,
+        targetAlignedPixels=True,
+        dstSRS='EPSG:4326',
+        format='GTiff'
+    )
 
-    del src_ds, dst_ds
+    flip_values_command = ' '.join([
```

Comment:
This could be a Python function like:
```python
def replace_geotiff_data(geotiff_path: str, stock_value: float) -> None:
    """Load a geotiff file, swap the non-zero data for all ones and resave."""
    ds = gdal.Open(geotiff_path, gdal.GA_Update)
    band = ds.GetRasterBand(1)
    data = band.ReadAsArray()
    data[data != 0 + 0j] = stock_value
    band.WriteArray(data)
    ds.FlushCache()
    ds = None
```

Reply:
I ended up having to add a bit more to that function. I feel like the CLI version is cleaner and more consistent with the rest of the code. Is there a significant benefit to this method? I compared the two in terms of speed and they're about the same.

```python
def switch_mask_values(input_image: str, output_image: str, gdal_format: str):
    """Swap the 1's and 0's in the provided mask.
    
    Args:
        input_image: 
    """
    ds = gdal.Open(input_image, gdal.GA_Update)
    band = ds.GetRasterBand(1)
    data = band.ReadAsArray()
    tmp = np.copy(data)
    data[tmp == 0 + 0j] = 1
    data[tmp != 0 + 0j] = 0
    band.WriteArray(data)
    ds.FlushCache()
    ds = None
    system(f'gdal_translate {input_image} {output_image} -of {gdal_format}')
```